From c6cfd44717643b0630b5f8fd3262c243fdaaf239 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Fri, 08 Mar 2013 09:54:45 +0000
Subject: Work around GCC 4.8 -Werror=format= in python3

...complainging that "‘PyArg_ParseTuple’ is an unrecognized format function
type."

Change-Id: I125af6669010c4c9c1a18cc7c1a4895acc89338b
---
diff --git a/python3/UnpackedTarball_python3.mk b/python3/UnpackedTarball_python3.mk
index d8c0862..87adff0 100644
--- a/python3/UnpackedTarball_python3.mk
+++ b/python3/UnpackedTarball_python3.mk
@@ -30,6 +30,7 @@ $(eval $(call gb_UnpackedTarball_add_patches,python3,\
 	python3/python-3.3.0-ssl.patch.1 \
 	python3/python-3.3.0-implicit-int.patch.1 \
 	python3/python-3.3.0-ffi-clang.patch.1 \
+	python3/python-3.3.0-gcc-4.8.patch.1 \
 ))
 
 ifneq ($(OS),WNT)
diff --git a/python3/python-3.3.0-gcc-4.8.patch.1 b/python3/python-3.3.0-gcc-4.8.patch.1
new file mode 100644
index 0000000..8fb7107
--- a/dev/null
+++ b/python3/python-3.3.0-gcc-4.8.patch.1
@@ -0,0 +1,15 @@
+--- Python-3.3.0/Include/modsupport.h	2012-09-29 10:00:26.000000000 +0200
++++ Python-3.3.0/Include/modsupport.h	2013-03-08 10:46:28.671938738 +0100
+@@ -26,7 +26,11 @@
+ /* Due to a glitch in 3.2, the _SizeT versions weren't exported from the DLL. */
+ #if !defined(PY_SSIZE_T_CLEAN) || !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x03030000
+ PyAPI_FUNC(int) PyArg_Parse(PyObject *, const char *, ...);
+-PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...) Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3);
++PyAPI_FUNC(int) PyArg_ParseTuple(PyObject *, const char *, ...)
++#ifdef PY_SSIZE_T_CLEAN
++    Py_FORMAT_PARSETUPLE(PyArg_ParseTuple, 2, 3)
++#endif
++    ;
+ PyAPI_FUNC(int) PyArg_ParseTupleAndKeywords(PyObject *, PyObject *,
+                                                   const char *, char **, ...);
+ PyAPI_FUNC(int) PyArg_ValidateKeywordArguments(PyObject *);
--
cgit v0.9.0.2-2-gbebe
